### PR TITLE
changed permissions to use cancancan gem

### PR DESCRIPTION
**Pull request by gsnovak**
cancancancancancancancancancancan gem @fiestacasey @austinkahly @ksolkowski

### CODE REVIEW

File: app/models/credit_card.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class CreditCard < ApplicationRecord
-  before_save :set_last_four
+  #this needs to be fixed before_save :set_last_four
```
Comment (gsnovak) [inline app/models/credit_card.rb]: just pretend this works for now

File: app/controllers/api/v1/base_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
-
 class Api::V1::BaseController < ApplicationController
+  load_and_authorize_resource
+
   before_action :retrieve_source_model_by_id, only: [:edit, :update, :destroy, :show]
```
Comment (fiestacasey) [inline app/controllers/api/v1/base_controller.rb:4]: I don't think you need this now that you are using cancancan. You should be able to refer to the model as `resource`

File: app/models/ability.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+      user ||= User.new
```
Comment (fiestacasey) [inline app/models/ability.rb]: how many spaces are you using?!?!?!?!!?

**Review by fiestacasey — CHANGES_REQUESTED** [review]

File: app/models/credit_card.rb [hunk 4]
```diff
@@ -9,6 +9,6 @@ class CreditCard < ApplicationRecord
   private
 
   def set_last_four
-    self.last_four = number[-4..-1]
+    self.last_four = number[-4..-1] if not number.nil?
```
Comment (fiestacasey) [inline app/models/credit_card.rb]: ```unless number.nil?```

**Pull request merged**